src/game/structures.js

```diff
@@ -1088,6 +1088,118 @@ exports.make = function(_runtimeData, _intents, _register, _globals) {
         return name;
     });
 
+    StructureSpawn.prototype.spawnCreep = register.wrapFn(function spawnCreep(body, name, options = {}) {
+
+        if(!name || !_.isObject(options)) {
+            return C.ERR_INVALID_ARGS;
+        }
+
+        if(globals.Game.creeps[name] || createdCreepNames.indexOf(name) != -1) {
+            return C.ERR_NAME_EXISTS;
+        }
+
+        let energyStructures = options.energyStructures && _.map(options.energyStructures, 'id');
+        let canResult = utils.canCreateCreep(runtimeData.roomObjects, this, {body, energyStructures});
+        if(options.dryRun || canResult !== C.OK) {
+            return canResult;
+        }
+
+        createdCreepNames.push(name);
+
+        if(_.isUndefined(globals.Memory.creeps)) {
+            globals.Memory.creeps = {};
+        }
+
+        if(_.isObject(globals.Memory.creeps)) {
+            if(_.isUndefined(options.memory)) {
```

Comment:
This if branch is unnecessary. If options.menory is undefined, it is falsey and will not be used. Just use the line in the else.

Reply:
Agreed. Adjusted in latest commit.